bmlt-meeting-list.php

```diff
@@ -26,36 +26,71 @@ class Bread {
+		const SETTINGS = 'bmlt_meeting_list_settings';
+		const OPTIONS_NAME = 'bmlt_meeting_list_options';
+		var $optionsName = Bread::OPTIONS_NAME;
 		var $options = array();
+		var $outside_meeting_result = array();
+		var $allSettings = array();
+		var $maxSetting = 1;
+		var $loaded_setting = 1;
+		
+		function loadAllSettings() {
+		    $this->allSettings = get_option( Bread::SETTINGS );
+		    if ($this->allSettings === false) {
+		        $this->allSettings = array();
+		        $this->allSettings[1] = "Default Setting";
+		        $this->maxSetting = 1;
+		    } else {
+		        foreach ($this->allSettings as $key => $value ) {
+		            if ($key > $this->maxSetting) {
+		                $this->maxSetting = $key;
+		            }
+		        }
+		    }
+		}
+		function startsWith($haystack, $needle)
+		{
+		    $length = strlen($needle);
+		    return (substr($haystack, 0, $length) === $needle);
+		}
 		function __construct() {
 		    $this->protocol = (strpos(strtolower(home_url()), "https") !== false ? "https" : "http") . "://";
-		    if (is_admin() || (isset($_GET['current-meeting-list']) && intval($_GET['current-meeting-list']) == 1)) {
-                $this->getMLOptions();
-                $this->lang = $this->get_bmlt_server_lang();
-
-                if (is_admin()) {
-                    // Back end
-                    //Initialize the options
-                    add_action("admin_init", array(&$this, 'my_sideload_image'));
-                    add_action("admin_menu", array(&$this, "admin_menu_link"));
-                    add_filter('tiny_mce_before_init', array(&$this, 'tiny_tweaks' ));
-                    add_filter('mce_external_plugins', array(&$this, 'my_custom_plugins'));
-                    add_filter('mce_buttons', array(&$this, 'my_register_mce_button'));
-                    add_action("admin_notices", array(&$this, "is_root_server_missing"));
-                    add_action("admin_init", array(&$this, "pwsix_process_settings_export"));
-                    add_action("admin_init", array(&$this, "pwsix_process_settings_import"));
-                    add_action("admin_init", array(&$this, "pwsix_process_default_settings"));
-                    add_action("admin_init", array(&$this, "my_theme_add_editor_styles"));
-                    add_action("admin_enqueue_scripts", array(&$this, "enqueue_backend_files"));
-                    add_action("wp_default_editor", array(&$this, "ml_default_editor"));
-                    add_filter('tiny_mce_version', array( __CLASS__, 'force_mce_refresh' ) );
-                } else if ( intval($_GET['current-meeting-list']) == 1 ) {
-                    $this->bmlt_meeting_list();
-                }
+			if (!($this->startsWith(strtolower($_SERVER['QUERY_STRING']),'current-meeting-list'))&&!is_admin() ) {
+		        return;
+		    }
+		    $this->loadAllSettings();
+		    $current_settings = 1;
+		    if (isSet($_GET) and isSet($_GET['current-meeting-list']))
```

Comment:
Any reason that you are using $_POST and $_COOKIE to get the parameter?

Reply:
You mean *not* using _POST?  Because it's convenient for the web pages where the PDF is downloaded to have a human readable link.  That's the way it's used now:  http://na?current-meeting-list=1

Reply:
querystrings are $_GET.  I see no reason to support cookies and post at this time.

Reply:
Oh, I see now I do use Cookie and Post...   I'm not sure with Cookie, that's just completeness.  The POST was to handle the Buttons "SAVE" and "GENERATE", which use html forms and post.

Reply:
The buttons are not using `<form/>` submit with `POST`.  They are just clicks to open URLs.

https://github.com/bmlt-enabled/bread/blob/4aa5585eb77018437e26f2658ecd74e63b55cc88/partials/_custom_section_setup.php#L45

Reply:
ok swap out $_GET and $_POST for $_REQUEST which means both.